Remotion/Web/ClientScript/Scripts/Utilities.ts

```diff
@@ -200,6 +200,52 @@ class ArgumentUtility
     ArgumentUtility.CheckNotNull(name, value);
     ArgumentUtility.CheckTypeIsJQueryObject(name, value);
   };
+
+  // Resolves any provided css selector, checks the element for null and returns it.
+  public static ResolveCssSelectorOrElement<TElement extends Element> (selectorOrElement: CssSelectorOrElement<TElement>, context?: ParentNode): TElement
```

Comment:
Um, you just added this to the existing ArgumentUtility class.

Reply:
Naming: `ResolveSingleElementOrCssSelector` and `ResolveUnboundedElementsOrCssSelector` or perhaps `ResolveMultipleElementsOrCssSelector`.  Reasoning: making it obvious if we're going for the single or the multiple variant.

There's still the question if `Resolve` is the best term for this. Do we resolve the element? I know we resolve the selector. But it just feels odd.

Reply:
Do you want the element resolving on another utility? I put it in the `ArgumentUtility` because it is used at the beginning of a method together with other argument checks:

```js
function test(selectorOrElement, key) {
  const element = ArgumentUtility.ResolveCssSelectorOrElement(selectorOrElement);
  ArgumentUtility.CheckIsString("key", key)
}
```

I know that it looks a little bit out of place, but since it also does argument checking I think it is a better solution than to have a Resolve utility and need to do argument checks after.

Reply:
I also do not like the naming 100%. It resolves the css selector part, otherwise (except from a null check) it returns the specified element.

Reply:
Resolving and argument checks should be separate tasks. Resolving should be a separate utility. => 2 calls at the start of the method. But then the resolve method could perhaps live without the argument check given we use typescript.

The argument check also tells you the name of the argument which failed, so this information would then be missing.

Reply:
Naming: we probably need to discuss some options to find the right one.

Reply:
Naming: I would suggest `ElementResolverUtility` with `ResolveSingleElementCssSelector` and `ResolveMultipleElementCssSelector` (note the missing "or" - otherwise the method it seems like it needs to resolve the element and not the css selector)

> But then the resolve method could perhaps live without the argument check given we use typescript.

I don't think I understand what you mean.

Reply:
Hmm...How about we just drop the CssSelector and make this `ResolveSingle` and `ResolveMultiple`. The `Element` we already have in the type so that's taken care of.

Reply:
> I don't think I understand what you mean.

As long as you use TypeScript, you can't pass a nullable argument value into the `Resolve` method. Therefor, we may not need to use argument checks inside the `Resolve` methods. We do use argument checks at the call sites because we get the css selectors passed down from c# and want to make sure nothing slips through.

Reply:
I have left the arg checks in the method, since they can catch some usages errors. Normally, the calling method does a null check before using the resolver, thus never causing the resolve method to throw. This prevents the resolve method from returning a nullable element, which would need an additional argument check on the call side.

so now:
```js
function test(selectorOrElement) {
  ArgumentUtility.CheckNotNull("selectorOrElement", selectorOrElement);
  const element = ElementResolve.ResolveSingle(selectorOrElement); // is always below the arg checks
  ...
}
```

Reply:
Added my review comment before I saw this mention.  Don't think my review contradicts the latest comment.